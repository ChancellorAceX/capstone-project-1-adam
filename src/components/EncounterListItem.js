import React from 'react';
import { Link } from 'react-router-dom';
import DeleteButton from './DeleteButton';

export default function EncounterListItem(props) {
  return (
    <Link to={`/encounter/${props.encounter.eid}`}>
      <li className='encounterListing'>
        {props.encounter.encountername}
        <DeleteButton
          target='encounter'
          id={props.encounter.eid}
          encounters={props.encounters}
          setTargetState={props.setTargetState}
        />
      </li>
    </Link>
  );
}