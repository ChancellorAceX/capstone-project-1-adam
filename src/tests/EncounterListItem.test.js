import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import EncounterListItem from '../components/EncounterListItem';

describe('EncounterListItem component', () => {
  const props = {
    "encounter": {
      "eid": 7,
      "encountername": "Green Fields",
      "encounterdesc": "Wolves",
      "encountercampaignid": 7,
      "cid": 7,
      "title": "Grishaverse",
      "created": "2020-07-03T00:23:10.591Z",
      "campaignuserid": 1
    },
    "encounters": [
      "{campaignuserid: 1, cid: 1, created: \"2020-07-03T00…}",
      "{campaignuserid: 1, cid: 6, created: \"2020-07-03T00…}",
      "{campaignuserid: 1, cid: 7, created: \"2020-07-03T00…}",
      "{campaignuserid: 1, cid: 1, created: \"2020-07-03T00…}",
      "{campaignuserid: 1, cid: 1, created: \"2020-07-03T00…}"
    ],
    "setTargetState": "ƒ () {}"
  };

  it('renders the full listing component', () => {
    const wrapper = shallow(<EncounterListItem {...props} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});