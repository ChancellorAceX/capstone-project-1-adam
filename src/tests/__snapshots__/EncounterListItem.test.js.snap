// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EncounterListItem component renders the full listing component 1`] = `
<Link
  to="/encounter/7"
>
  <li
    className="encounterListing"
  >
    Green Fields
    <DeleteButton
      encounters={
        Array [
          "{campaignuserid: 1, cid: 1, created: \\"2020-07-03T00…}",
          "{campaignuserid: 1, cid: 6, created: \\"2020-07-03T00…}",
          "{campaignuserid: 1, cid: 7, created: \\"2020-07-03T00…}",
          "{campaignuserid: 1, cid: 1, created: \\"2020-07-03T00…}",
          "{campaignuserid: 1, cid: 1, created: \\"2020-07-03T00…}",
        ]
      }
      id={7}
      setTargetState="ƒ () {}"
      target="encounter"
    />
  </li>
</Link>
`;
